packages/grid/x-data-grid/src/DataGrid/DataGrid.tsx

```diff
@@ -640,28 +634,11 @@ DataGridRaw.propTypes = {
   /**
    * Sets the row selection model of the grid.
    */
-  rowSelectionModel: chainPropTypes(
-    PropTypes.oneOfType([PropTypes.number, PropTypes.string, PropTypes.array]),
-    (props: any) => {
-      if (
-        !props.checkboxSelection &&
-        Array.isArray(props.rowSelectionModel) &&
-        props.rowSelectionModel.length > 1
-      ) {
-        return new Error(
```

Comment:
I've removed this error - `setRowSelectionModel` now throws an error when there's more than 1 item in `DataGrid` (since `apiRef` is now available in Community plan).
By the way, this was just a warning - multi row selection still worked: https://codesandbox.io/s/infallible-bhaskara-f0tbk4?file=/demo.tsx
So it wasn't really limited. Now it is.

Reply:
This check just printed an error message in the console, but the multi-row selection still worked in the Community package.
I've moved the check to the `setRowSelectionModel` method instead, so the error will be thrown when using both the method itself and the prop: https://github.com/mui/mui-x/blob/d52e4213ee99482d1ca73daac72524afb419f4fa/packages/grid/x-data-grid/src/hooks/features/rowSelection/useGridRowSelection.ts#L151-L161